sensors/simsensor_common/simsensor_generic.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Generic functions for all sensor simulators
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "simsensor_generic.h"
+
+#include "event_queue.h"
+#include "simsensor_reader.h"
+
+#include <stdlib.h>
+#include <errno.h>
+#include <time.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/time.h>
+#include <sys/threads.h>
+
+
+typedef struct {
+	char name[SIMSENS_NAME_MAX_LEN + 1];
+	event_queue_t eventQueue;
+	simsens_reader_t reader;
+	char stack[512] __attribute__((aligned(8)));
+} simsens_ctx_t;
+
+
+static void simsens_freeCtx(simsens_ctx_t *ctx)
+{
+	eventQueue_free(&ctx->eventQueue);
+	reader_close(&ctx->reader);
+	free(ctx);
+}
+
+
+static void simsens_publishthr(void *data)
+{
+	sensor_info_t *info = (sensor_info_t *)data;
+	simsens_ctx_t *ctx = info->ctx;
+	sensor_event_t event;
+	time_t now;
+
+	while (1) {
```

Comment:
This loop is not truly infinite. It has non-erroneous possibility termination possibility when `READER_STOP_EVENT` is found. Please make this loop condition rely on this. It may be just a boolean flag.